### ISSUE TITLE
this._mempool.on is not a function

### ISSUE THREAD

**Issue by neptunix** [Sushipool/source#1]
```sushipool_1  | [I 05:42:45] SushiPool: Reading config from argv
sushipool_1  | [I 05:42:46] SushiPool: Sushipool Miner 0.3.2 starting
sushipool_1  | [I 05:42:46] SushiPool: - network          = main
sushipool_1  | [I 05:42:46] SushiPool: - no. of threads   = 2
sushipool_1  | [I 05:42:46] SushiPool: - pool server      = eu.sushipool.com:443
sushipool_1  | [I 05:42:46] SushiPool: - address          = NQ18 ...
sushipool_1  | [I 05:42:46] SushiPool: - device name      = mb
sushipool_1  | [I 05:42:46] SushiPool: Please wait while we establish consensus.
sushipool_1  | [I 05:42:47] SushiPool: Wallet initialized for address NQ18 ... . Balance: 0 NIM
sushipool_1  | [I 05:42:47] SushiPool: Blockchain state: height=1, headHash=Jkqvik+YKKdsVQY12geOtGYwahifzANxC+6fZJyGnRI=
sushipool_1  | TypeError: this._mempool.on is not a function
sushipool_1  |     at new Miner (/home/root/miner/miner/node_modules/@nimiq/core/dist/node.js:34990:27)
sushipool_1  |     at new BasePoolMiner (/home/root/miner/miner/node_modules/@nimiq/core/dist/node.js:35337:9)
sushipool_1  |     at new SushiPoolMiner (/home/root/miner/miner/src/SushiPoolMiner.js:19:9)
sushipool_1  |     at /home/root/miner/miner/index.js:143:15
sushipool_1  |     at <anonymous>
sushipool_1  |     at process._tickCallback (internal/process/next_tick.js:182:7)
sushipool_1  |     at Function.Module.runMain (internal/modules/cjs/loader.js:697:11)
sushipool_1  |     at startup (internal/bootstrap/node.js:201:19)
sushipool_1  |     at bootstrapNodeJSCore (internal/bootstrap/node.js:516:3)
```

Adding some debug to `/home/root/miner/miner/node_modules/@nimiq/core/dist/node.js` it seems that params come in wrong order:

```
Blockchain=> Accounts {
  _listeners: Map {},
  _tree:
   AccountsTree {
     _listeners: Map { '*' => [Array] },
     _store: AccountsTreeStore { _store: [ObjectStore] },
     _synchronizer:
      Synchronizer {
        _listeners: Map {},
        _queue: [LinkedList],
        _working: false,
        _throttleAfter: undefined,
        _throttleWait: undefined,
        _elapsed: 0,
        _totalElapsed: 0,
        _totalJobs: 0,
        _totalThrottles: 0 } } } 

Accounts=> Mempool {
  _listeners:
   Map {
     'transaction-added' => [ [Function] ],
     'transaction-removed' => [ [Function] ] },
  _blockchain:
   LightChain {
     _listeners: Map { 'head-changed' => [Array], 'rebranched' => [Array] },
     _store:
      ChainDataStore { _chainStore: [ObjectStore], _blockStore: [ObjectStore] },
     _accounts: Accounts { _listeners: Map {}, _tree: [AccountsTree] },
     _time: Time { _offset: 0 },
     _snapshots: HashMap { _map: Map {}, _fnHash: [Function: _hash] },
     _snapshotOrder: [],
     _mainChain:
      ChainData {
        _head: [Block],
        _totalDifficulty: [BigNumber],
        _totalWork: [BigNumber],
        _superBlockCounts: [SuperBlockCounts],
        _onMainChain: true,
        _mainChainSuccessor: null,
        _height: 1 },
     _proof: ChainProof { _prefix: [BlockChain], _suffix: [HeaderChain] },
     _transactionCache:
      TransactionCache { _transactionHashes: [InclusionHashSet], _blockOrder: [Array] },
     _transactionStore: undefined,
     _synchronizer:
      PrioritySynchronizer {
        _listeners: Map {},
        _queues: [Array],
        _working: false,
        _throttleAfter: 500,
        _throttleWait: 30,
        _elapsed: 0,
        _totalElapsed: 0,
        _totalJobs: 0,
        _totalThrottles: 0 },
     _blockForkedCount: 0,
     _blockRebranchedCount: 0,
     _blockExtendedCount: 0,
     _blockOrphanCount: 0,
     _blockInvalidCount: 0,
     _blockKnownCount: 0,
     _headHash: Hash { _obj: [SerialBuffer], _algorithm: 1 } },
  _accounts:
   Accounts {
     _listeners: Map {},
     _tree:
      AccountsTree {
        _listeners: [Map],
        _store: [AccountsTreeStore],
        _synchronizer: [Synchronizer] } },
  _transactionsByFeePerByte: SortedList { _list: [], _compare: [Function: _compare] },
  _transactionsByHash: HashMap { _map: Map {}, _fnHash: [Function: _hash] },
  _transactionSetBySender: HashMap { _map: Map {}, _fnHash: [Function: _hash] },
  _transactionSetByRecipient: HashMap { _map: Map {}, _fnHash: [Function: _hash] },
  _synchronizer:
   Synchronizer {
     _listeners: Map {},
     _queue: LinkedList { _length: 0, _head: null, _tail: null },
     _working: false,
     _throttleAfter: undefined,
     _throttleWait: undefined,
     _elapsed: 0,
     _totalElapsed: 0,
     _totalJobs: 0,
     _totalThrottles: 0 } } 

Mempool=> Time { _offset: 0 } 

Time=> Address {
  _obj:
   Uint8Array [
     133,
     136,
     ...    ] } 

MinerAddress=> 2627988788
TypeError: this._mempool.on is not a function
```

nimiq.core version": "1.2.0"

**Comment by neptunix** [comment]
Steps to reproduce: :)
* git clone
* cd source
* npm i
* ./sushipool

**Comment by brantje** [comment]
Hi,
Thanks for reporting, since the problem is in `@nimiq/core` it's better to report it [upstream](https://github.com/nimiq-network/core).

Thanks in advance!

**Comment by sisou** [comment]
@brantje This is in fact a bug in `ShushiPoolMiner`, because an extra paramter was added to `BasePoolMiner` here: https://github.com/nimiq-network/core/commit/38d8b9671d6618428c59030ad844d70c28186b92#diff-3534365a792ca2e9a326d2b4e9bba56eR15

It now expects the `mode` (`'nano'` or `'smart'`) as the first parameter, shifting all following parameters.

**Comment by brantje** [comment]
Fixed in 9d3bf8ce5a399717d1829c779cc466f52050e216
